### DISCUSSION TITLE
Negative number / JSDoc how-to

### DISCUSSION THREAD

**Discussion by rolanday** [sinclairzx81/typebox-codegen#22] [General]
Hello,

This is prob a JSDoc question but no luck finding an answer. I want to express lower bound as negative like so:

```
// Edit TypeScript Types Here. Use Ctrl + S to Share

export type T = {
  /**
   * @minimum -90
   * @maximum 90
   */
  x: number
}
```
However, codegen returns the following when I do:

```
import { Type, Static } from '@sinclair/typebox'

export type T = Static<typeof T>
export const T = Type.Object({
  x: Type.Number({ minimum: '', maximum: 90 }) // minimum is interpreted as an empty string
})

```

How do I mark-up JSDoc param so minimum is parsed correctly?

Any help much appreciated. Thanks! :-)

**Comment by sinclairzx81** [comment]
@rolanday Hi,

It's a little unintuitive, but you can quote the negative value.

```typescript
export type T = {
  /**
   * @minimum "-90"
   * @maximum "90"
   */
  x: number
}
```
Results in 
```typescript
import { Type, Static } from '@sinclair/typebox'

export type T = Static<typeof T>
export const T = Type.Object({
  x: Type.Number({ minimum: -90, maximum: 90 })
})
```
This project implements a custom JSDoc parser which encodes the rule that if a quoted value appears to be numeric, it will be converted into a number. Also, the `-` is observed as a parameter comment delimiter in JSDoc, so quoting the delimiter is a way to get around this. I may look at better JSDoc parser in the new year.

Hope this helps
S

**Reply by rolanday** [reply]
@sinclairzx81 , yes, very helpful (quoting did the trick). Thanks so much for the super quick response and, again, for the fantastic contrib -- really enjoying working with it.
